model/textparse/openmetricsparse.go

```diff
@@ -260,25 +263,24 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
 	if !typeRequiresCT(p.mtype) {
 		// Not a CT supported metric type, fast path.
-		p.ct = 0
-		p.visitedName = ""
-		p.ctHashSet = 0
+		p.ctHashSet = 0 // Use ctHashSet as a single way of telling "empty cache"
 		return nil
 	}
 
 	var (
-		currLset                labels.Labels
-		buf                     []byte
-		peekWithoutNameLsetHash uint64
+		buf      []byte
+		currName []byte
 	)
-	p.Metric(&currLset)
-	currFamilyLsetHash, buf := currLset.HashWithoutLabels(buf, labels.MetricName, "le", "quantile")
-	currName := currLset.Get(model.MetricNameLabel)
-	currName = findBaseMetricName(currName)
-
-	// make sure we're on a new metric before returning
-	if currName == p.visitedName && currFamilyLsetHash == p.ctHashSet && p.visitedName != "" && p.ctHashSet > 0 && p.ct > 0 {
-		// CT is already known, fast path.
+	if p.series[0] == '{' && p.series[1] == '"' {
```

Comment:
suspect. Unguarded array indexing. Maybe? Could we have single letter metric name? Please try test case.
```suggestion
	if len(p.series)>1 && p.series[0] == '{' && p.series[1] == '"' {
```

Reply:
true but this would be a very weird case where metric name is just `{` and im not sure how to reach this sorta test case.
I'll add it in tho